JASP-Engine/JASP/R/correlation.R

```diff
@@ -15,806 +15,721 @@
+  reportCI=FALSE, reportVovkSellkeMPR=FALSE, flagSignificant=FALSE) {
 
 	correlation.table <- list()
-
-	numberOfVariables <- length(variables)
-	
-	if (perform == "init") {
-	    if (numberOfVariables == 0){
-	        variables <- c(variables, "...", "... ")
-	    } else if (numberOfVariables == 1){
-	        variables <- c(variables, "... ")
-	    }
-	}
-	
-	# update number of variables, hence it's always >= 2, thus, outputting a table of 2 by 2
-	#
-	numberOfVariables <- length(variables)
-	
-	tests <- c()
-	if (pearson)
-		tests <- c(tests, "pearson")
-	if (spearman)
-		tests <- c(tests, "spearman")
-	if (kendallsTauB)
-		tests <- c(tests, "kendall")
-
+  
+	tests <- tableData$tests
 	if (length(tests) != 1) {
-
 		correlation.table[["title"]] <- "Correlation Table"
-	}
-	else if (pearson) {
-
+	} else if (tests == "pearson") {
 		correlation.table[["title"]] <- "Pearson Correlations"
-	}
-	else if (spearman) {
-
+	} else if (tests == "spearman") {
 		correlation.table[["title"]] <- "Spearman Correlations"
-	}
-	else if (kendallsTauB) {
-
-		correlation.table[["title"]] <- "Kendall's Tau"
-	}
-	else {
-
+	} else if (tests == "kendall") {
+		correlation.table[["title"]] <- "Kendall's Tau Correlations"
+	} else {
 		correlation.table[["title"]] <- "Correlation Table"
 	}
-
-	fields <- list(list(name=".variable", title="", type="string"))
-	rows <- list()
-
+  
+  hypothesis <- tableData$hypothesis
 	footnotes <- .newFootnotes()
-
 	if (flagSignificant || reportSignificance) {
 
 		if (hypothesis == "correlatedPositively") {
-
 			.addFootnote(footnotes, "all tests one-tailed, for positive correlation", symbol="<i>Note</i>.")
-
 		} else if (hypothesis == "correlatedNegatively") {
-
 			.addFootnote(footnotes, "all tests one-tailed, for negative correlation", symbol="<i>Note</i>.")
 		}
+    
 	}
 
 	if (flagSignificant) {
 
 		if (hypothesis == "correlated") {
-
 			.addFootnote(footnotes, "p < .05, ** p < .01, *** p < .001", symbol="*")
-
 		} else {
-
 			.addFootnote(footnotes, "p < .05, ** p < .01, *** p < .001, one-tailed", symbol="*")
 		}
+    
 	}
 
 	if (reportVovkSellkeMPR) {
+    
 		if (flagSignificant){
 	    .addFootnote(footnotes, symbol = "\u207A", text = "Vovk-Sellke Maximum
-	    <em>p</em>-Ratio: Based the <em>p</em>-value, the maximum
+	    <em>p</em>-Ratio: Based on the <em>p</em>-value, the maximum
 	    possible odds in favor of H\u2081 over H\u2080 equals
 	    1/(-e <em>p</em> log(<em>p</em>)) for <em>p</em> \u2264 .37
 	    (Sellke, Bayarri, & Berger, 2001).")
 		} else {
 			.addFootnote(footnotes, symbol = "\u002A", text = "Vovk-Sellke Maximum
-	    <em>p</em>-Ratio: Based the <em>p</em>-value, the maximum
+	    <em>p</em>-Ratio: Based on the <em>p</em>-value, the maximum
 	    possible odds in favor of H\u2081 over H\u2080 equals
 	    1/(-e <em>p</em> log(<em>p</em>)) for <em>p</em> \u2264 .37
 	    (Sellke, Bayarri, & Berger, 2001).")
 		}
+    
 	}
+  
+  variables <- tableData$variables
+  CI <- tableData$CI
```

Comment:
this is a local CI isn't it? OR is this the list of all CIs?

Reply:
Global CI. tableData contains everything needed to populate results in the table.